spec/unit/facter/puppet_settings.rb

```diff
@@ -0,0 +1,27 @@
+require 'spec_helper'
+# not sure how best to predict this? the following won't work on windows
+hostname = `hostname -f`.strip
+settings = {
+  "main" => {
+    "hostcert" => "/dev/null/ssl/certs/#{hostname}.pem",
+    "hostprivkey" => "/dev/null/ssl/private_keys/#{hostname}.pem",
+    "hostpubkey" => "/dev/null/ssl/public_keys/#{hostname}.pem",
+    "localcacert" => '/dev/null/ssl/certs/ca.pem',
+    "server" => 'puppet',
+    "ssldir" => '/dev/null/ssl',
+    "vardir" => '/dev/null',
+  },
+  "master" => {
+    "localcacert" => '/dev/null/ssl/certs/ca.pem',
+    "ssldir" => '/dev/null/ssl',
+  }
+}
+describe 'puppet_config' do
+  let(:node) { 'puppet_config.example.org' }
```

Comment:
Underscores in hostnames are not allowed

Reply:
This didn;t actully work and should have been removed, thanks